herd/event.ml

```diff
@@ -1311,10 +1311,11 @@ module Make  (C:Config) (AI:Arch_herd.S) (Act:Action.S with module A = AI) :
       let r =
         { r with
           intra_causality_order =
+            let f e = (is_explicit e) || (is_fault e) in
```

Comment:
As far as I have checked `STP` does not order register reads nor memory writes. Or am I missing something?

Reply:
Here, I am only changing `seq_mem` but I think in a way that makes sense.

`seq_mem` is used only in herd/AArch64Sem.ml in the functions `ldp`, `ldp_wback`, `stp` and `stp_wback`. However it's not used for the semantics of all `LDP` and `STP` variants. It's only used to build the semantics for `LDIAPP` (in all addressing modes, indexed, pre-indexed and post-indexed) and `STILP` (also in all addressing modes). The semantics of `LDIAPP` and `STILP` require that their accesses are ordered. And that's not something that this PR changes. It won't order the accesses of other pair instructions (e.g. `LDP`, `LDNP`, `STP`, etc.).

Also, I think the change to `seq_mem` makes sense and I think the change is in the spirit of the original intent. For instructions such as `LDIAPP` and `STILP` the intent is to order their two explicit accesses only. I don't think the intent was to order their implicit memory accesses (for example, the Implicit PTE Memory Reads, Implicit PTE Memory Writes, etc).

But if you think otherwise, then I am more than happy to revert this. As it happens, we don't need to use it in the implementation of STZG so maybe I will revert it anyway (I'll keep a revised version of the comment in `monad.mli` if that's ok with you)

Reply:
I've now reverted this change because as I mentioned, I ended up having to change the semantics of `STZG` after some feedback and I am not using `seq_mem`. So maybe it's not worth making this change here. I didn't drop the change to the comment in `monad.mli` though because hopefully that's useful nevertheless.